chapter_02/lewis/exercise_1.pl

```diff
@@ -0,0 +1,18 @@
+###
+# (C) 2017 Lewis Maitland
+# O'Reily Excersises
+###
+
+use strict;
+use warnings;
+use Math::Trig ':pi';
+
+my $radius = $#ARGV == 0 ? $ARGV[0] : 12.5;
+
+if( $radius !~ /^[0-9]+(\.[0-9]+)?$/ ) {
+	print "Invalid radius supplied.\n";
+	exit 0;

```

Comment:
``exit 0`` implies the script ended successfully, which the message doesn't seem to imply. If you want to exit a script unsuccessfully use ``die``. If you want to exit early but successfully, use ``exit`` without specifying an exit code. It's best to leave the exit codes alone unless you know what you're doing with them.